core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -97,27 +97,37 @@ public String name() {
 
   @Override
   public RewriteStrategy options(Map<String, String> options) {
-    targetFileSize = PropertyUtil.propertyAsLong(options,
-        RewriteDataFiles.TARGET_FILE_SIZE_BYTES,
-        PropertyUtil.propertyAsLong(
-            table().properties(),
-            TableProperties.WRITE_TARGET_FILE_SIZE_BYTES,
-            TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT));
-
-    minFileSize = PropertyUtil.propertyAsLong(options,
-        MIN_FILE_SIZE_BYTES,
+    Map<String, String> tableProperties = table().properties();
+
+    targetFileSize = PropertyUtil.resolveLongProperty(
```

Comment:
This ends up being pretty difficult to read for me. I wonder if we should have this similar to the way set up SparkReadConf and SparkWriteConf options. Maybe ActionConf?

Reply:
After looking through "PropertyUtil.resolve*" I think It may be much clearer if we use specified methods like in SparkReadConf and SparkWriteConf, so this would start with something like ActionsConf(action, table, options) or what not. And then you would call conf.targetFileSize().

Reply:
ActionsConf sounds good to me, thanks for the suggestion, let me update the PR